pkg/hive/cell/metric.go

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cell
+
+import (
+	"fmt"
+	"reflect"
+	"runtime"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/dig"
+
+	"github.com/cilium/cilium/pkg/hive/internal"
+	pkgmetric "github.com/cilium/cilium/pkg/metrics/metric"
+)
+
+func Metric[S any](ctor func() S) Cell {
```

Comment:
Given all the runtime constraints on the passed constructor, might be worth documenting this in the function comment.